semantic_model_generator/generate_model.py

```diff
@@ -146,6 +147,18 @@ def _raw_table_to_semantic_context_table(
             f"No valid columns found for table {raw_table.name}. Please verify that this table contains column's datatypes not in {OBJECT_DATATYPES}."
         )
 
+    primary_key = None
+    if allow_joins:
+        # Populate the primary key field if we were able to retrieve one during raw table construction.
+        # If not, leave a placeholder for the user to fill out.
+        primary_key = semantic_model_pb2.PrimaryKey(
+            columns=(
+                raw_table.primary_key
```

Comment:
we currently have a constraint where the primary key must be defined using logical columns.
are you working under the assumption that the OSS generator is also naming logical columns the same as physical columns?

Reply:
ah gotcha, I wasn't aware of that constraint. yes I think I'm operating under that assumption; I'd presume that doing `show primary key in table...` returns physical column names - do you have a recommendation on how we'd do a mapping if necessary?

Reply:
you run this function only the first time someone is creating a new semantic model right? it is not run a second time after people change the name of columns?

i guess the fallback would be, if they changed the name of the columns, our validation logic (in orchestrator) will detect it and return an error.

the longer way would be to a mapping step on yaml change. but i dont think it is worth your time at this point

Reply:
Validation is run both 1) before the user uploads the YAML to a stage, and 2) whenever they attempt to chat. So if we have validation logic in the orchestrator to catch this, we should be good to warn the user before they're finished saving the model to stage.